lib/pbench/server/cache_manager.py

```diff
@@ -192,29 +181,115 @@ def make_cache_object(dir_path: Path, path: Path) -> CacheObject:
     )
 
 
+class LockRef:
+    """Keep track of a cache lock passed off to a caller"""
+
+    def __init__(self, lock: Path, exclusive: bool = False, wait: bool = True):
```

Comment:
Instead of implementing `LockRef` _as_ a context manager, consider instead implementing a function which wraps it in a context manager using [`contextlib.contextmanager`](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager).  The uses might look something like
```python
with tarball.with_lock(exclusive=True, wait=False, keep=False) as lock:
```
Where `Tarball.with_lock()` would look like:
```python
    @contextlib.contextmanager
    def with_lock(self, exclusive=True, wait=False, keep=False):
        lock = LockRef(self.lock)
        try:
            lock.acquire(exclusive, wait)
            yield lock
        finally:
            if not keep:
                lock.release()
```
I think this frees us to unburden the c'tor and the `LockRef` class from several of its attributes, because we can instead pass them as parameters where and when they are needed.

Reply:
(BTW, there are several cool looking widgets in `contextlib` -- of particular note are [`suppress`](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) and [`ExitStack`](https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack).)

Reply:
> (BTW, there are several cool looking widgets in `contextlib` -- of particular note are [`suppress`](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) and [`ExitStack`](https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack).)

`ExitStack` could have served the purpose for my intake unwind stack; it's a bit more general (if also a bit more cumbersome) than the one I invented. Hardly worth worrying about, though.

Reply:
I wasn't suggesting using `suppress` or `ExitStack` in this PR...I just found them after finding the `contextmanager` decorator and figured I would mention them, as they nicely handle things that we do now and then.